### PR TITLE
Avoid crashing Alchemy when there is no network or The Graph servers are overloaded - 2.0

### PR DESCRIPTION
**Pull request by roienatan**
resolves: https://github.com/daostack/alchemy/issues/2231 for 2.0

### CODE REVIEW

**Review by orenyodfat — CHANGES_REQUESTED** [review]
it does not seems to continue polling after the network error

**Comment by roienatan** [comment]
> it does not seems to continue polling after the network error

Looks like React Apollo Client does not resume polling after a network error, see this:
https://github.com/apollographql/apollo-client/issues/1099
I need to investigate and see if there is an option to resume the polling. The fix might be in arc.js if so.

File: src/arc.ts [hunk 1]
```diff
@@ -124,21 +123,14 @@ export async function initializeArc(provider?: Web3Provider): Promise<boolean> {
...
-        max: 5,
-        retryIf: (error, _operation) => {
-          // eslint-disable-next-line no-console
-          console.error("error occurred fetching data, retrying...");
-          // eslint-disable-next-line no-console
-          console.log(error);
-          return !!error;
-        },
+      attempts: (count) => {
+        if (count === 10) {
```
Comment (orenyodfat) [inline src/arc.ts:127]: this lines can be changed to : 
`return (count !== 10);`

File: src/arc.ts [hunk 2]
```diff
@@ -124,21 +123,14 @@ export async function initializeArc(provider?: Web3Provider): Promise<boolean> {
...
+        }
+        return true;
       },
-      delay: {
-        initial: 500, // this is the initial time after the first retry
-        // next retries )up to max) will be exponential (i..e after 2*iniitial, etc)
-        jitter: true,
-        max: Infinity,
+      delay: () => {
+        return Math.floor(Math.random() * (30000 - 5000 + 1) + 5000);
```
Comment (orenyodfat) [inline src/arc.ts:133]: lets add a comment here : 
" this will give a random delay between retries between the range of 5 to 30 seconds"

**Review by orenyodfat — APPROVED** [review]

**Pull request merged**